src/main/credentialServer/httpCredentialHelperServer.ts

```diff
@@ -143,43 +183,76 @@ export class HttpCredentialHelperServer {
     }
     if (requestCheckError) {
-      response.writeHead(400, { 'Content-Type': 'text/plain' });
-      response.write(requestCheckError);
+      throw new Error(requestCheckError);
+    }
 
-      return;
+    const platform = os.platform();
+    let pathVar = process.env.PATH ?? '';
+
+    // The PATH needs to contain our resources directory (on macOS that would
+    // not be in the application's PATH), as well as /usr/local/bin.
+    // NOTE: This needs to match DockerDirManager.spawnFileWithExtraPath
+    pathVar += path.delimiter + path.join(paths.resources, platform, 'bin');
+    if (platform === 'darwin') {
+      pathVar += `${ path.delimiter }/usr/local/bin`;
     }
 
-    try {
-      const platform = os.platform();
-      let pathVar = process.env.PATH ?? '';
-
-      // The PATH needs to contain our resources directory (on macOS that would
-      // not be in the application's PATH), as well as /usr/local/bin.
-      // NOTE: This needs to match DockerDirManager.
-      pathVar += path.delimiter + path.join(paths.resources, platform, 'bin');
-      if (platform === 'darwin') {
-        pathVar += `${ path.delimiter }/usr/local/bin`;
-      }
+    const body = stream.Readable.from(data);
+    const { stdout } = await childProcess.spawnFile(helperName, [commandName], {
+      env:   { ...process.env, PATH: pathVar },
+      stdio: [body, 'pipe', console],
+    });
 
-      const body = stream.Readable.from(data);
-      const { stdout } = await childProcess.spawnFile(helperName, [commandName], {
-        env:   { ...process.env, PATH: pathVar },
-        stdio: [body, 'pipe', console],
-      });
+    if (!checkerFn(stdout)) {
+      throw new Error(`Invalid output for ${ commandName } command.`);
+    }
 
-      if (checkerFn(stdout)) {
-        response.writeHead(200, { 'Content-Type': 'text/plain' });
-        response.write(stdout);
+    return stdout;
+  }
 
-        return;
+  /**
+   * For the LIST command, there are multiple possible sources of information
+   * that need to be merged into a simple
+   *    { ServerURL: Username } hash.
+   * The first source is the credsStore.
+   * Then if any helper credsStores are identified in the `credHelpers` section,
+   * get the full { ServerURL: Username } from each of them,
+   * and keep only those ServerURLs that point to that credsStore.
+   *
+   * Modeled after https://github.com/docker/cli/blob/d0bd373986b6678bfe1a0eb6989ce13907247a85/cli/config/configfile/file.go#L285
+   */
+  protected async doListCommand(
+    thisHelperInfo: credHelperInfo,
+    request: http.IncomingMessage,
+    response: http.ServerResponse): Promise<void> {
+    try {
+      const serverAndUsernameInfo: Record<string, string> = JSON.parse(await this.runCommand(`docker-credential-${ thisHelperInfo.credsStore }`, 'list', '', request));
+      const names = _.uniq(Object.values(thisHelperInfo.credHelpers ?? {}));
```

Comment:
```suggestion
      const names = new Set(Object.values(thisHelperInfo.credHelpers ?? {}));
```
(A set is iterable, so the other code doesn't need to change.)

This lets us drop the import.

Reply:
Sure